v3/src/models/shared/shared-case-metadata.ts

```diff
@@ -194,6 +194,31 @@ export const SharedCaseMetadata = SharedModel
       return categorySet
     }
   }))
+  .actions(self => ({
+    afterCreate() {
+      // respond to change of collections by showing hidden attributes if all attributes are hidden
+      addDisposer(self, reaction(
```

Comment:
This should be an `mstReaction` on `self.data`, so that if the `DataSet` is destroyed before the `SharedCaseMetadata`, the reaction will be disposed of before firing after the `DataSet` is defunct.

Reply:
I tried this locally and it's not as straightforward as I'd hoped, so let's skip it for now.